zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +159,253 @@ def compute(self, today, assets, out, close, volume):
         out[:] = nanmean(close * volume, axis=0)
 
 
+class _RollingCorrelationOfReturns(CustomFactor, SingleInputMixin):
+    """
+    Base class for factors computing a rolling statistical measure over a
+    window of Returns.
+
+    Parameters
+    ----------
+    baseline_asset : zipline.assets.Asset
+        The asset to correlate with all other assets.
+    returns_length : int > 1
+        Length of the lookback window over which to compute returns.
+    correlation_length : int > 1
+        Length of the lookback window over which to compute the desired
+        statistical measure.
+    """
+    params = ['baseline_asset']
+
+    def __new__(cls,
+                baseline_asset,
+                returns_length,
+                correlation_length,
+                mask=NotSpecified,
+                **kwargs):
+        if mask is not NotSpecified:
+            # Make sure we do not filter out the asset of interest.
+            mask = mask | SingleAsset(asset=baseline_asset)
+        return super(_RollingCorrelationOfReturns, cls).__new__(
+            cls,
+            baseline_asset=baseline_asset,
+            inputs=[Returns(window_length=returns_length)],
+            window_length=correlation_length,
+            mask=mask,
+            **kwargs
+        )
+
+
+class RollingPearsonOfReturns(_RollingCorrelationOfReturns):
+    """
+    Calculates the Pearson product-moment correlation coefficient of the
+    returns of the given asset with the returns of all other assets. Note
```

Comment:
I would separate this second sentence from the first with a blank line so it's harder to miss.  I'd also probably drop the `Note that` in favor of just: "Pearson correlation is what most people mean when they say "correlation coefficient" or "R-value".